crates/store/benches/account_tree_historical.rs

```diff
@@ -0,0 +1,239 @@
+use std::hint::black_box;
+
+use criterion::{BenchmarkId, Criterion, criterion_group, criterion_main};
+use miden_node_store::{AccountTreeWithHistory, InMemoryAccountTree};
+use miden_objects::Word;
+use miden_objects::account::AccountId;
+use miden_objects::block::BlockNumber;
+use miden_objects::block::account_tree::{AccountTree, account_id_to_smt_key};
+use miden_objects::crypto::hash::rpo::Rpo256;
+use miden_objects::crypto::merkle::{LargeSmt, MemoryStorage};
+use miden_objects::testing::account_id::AccountIdBuilder;
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Creates a storage backend for a `LargeSmt`.
+fn setup_storage() -> MemoryStorage {
+    // TODO migrate to RocksDB for persistence to gain meaningful numbers
+    MemoryStorage::default()
+}
+
+/// Generates a deterministic word from a seed.
+fn generate_word(seed: &mut [u8; 32]) -> Word {
+    for byte in seed.iter_mut() {
+        *byte = byte.wrapping_add(1);
+        if *byte != 0 {
+            break;
+        }
+    }
+    Rpo256::hash(seed)
+}
+
+/// Generates a deterministic `AccountId` from a seed.
+fn generate_account_id(seed: &mut [u8; 32]) -> AccountId {
+    for byte in seed.iter_mut() {
+        *byte = byte.wrapping_add(1);
+        if *byte != 0 {
+            break;
+        }
+    }
+    AccountIdBuilder::new().build_with_seed(*seed)
+}
+
+// SETUP FUNCTIONS
+// ================================================================================================
+
+/// Sets up a vanilla `AccountTree` with specified number of accounts using `RocksDB` backend.
+fn setup_vanilla_account_tree(
+    num_accounts: usize,
+) -> (AccountTree<LargeSmt<MemoryStorage>>, Vec<AccountId>) {
+    let mut seed = [0u8; 32];
+    let mut account_ids = Vec::new();
+    let mut entries = Vec::new();
+
+    for _ in 0..num_accounts {
+        let account_id = generate_account_id(&mut seed);
+        let commitment = generate_word(&mut seed);
+        account_ids.push(account_id);
+        entries.push((account_id_to_smt_key(account_id), commitment));
+    }
+
+    let storage = setup_storage();
+    let smt =
+        LargeSmt::with_entries(storage, entries).expect("Failed to create LargeSmt from entries");
+    let tree = AccountTree::new(smt).expect("Failed to create AccountTree");
+    (tree, account_ids)
+}
+
+/// Sets up `AccountTreeWithHistory` with specified number of accounts and blocks using `RocksDB`
```

Comment:
nit: the RocksDB storage backend part looks misleading, but this seems like it's going to change soon anyway